app.py

```diff
@@ -227,7 +227,15 @@ def index(category=[]):
 def tag_index(slug):
     api_url = ''.join([INSIGHTS_URL, '/tags?slug=', slug])
     response = _get_from_cache(api_url)
-    tag = json.loads(response.text)[0]
+
+    response_text = json.loads(response.text)
+    if response_text:
+        tag = response_text[0]
+    else:
+        tag = {
+            "id": -1,
+            "name": slug
+        }
 
     page = flask.request.args.get('page')
     posts, metadata = _get_posts(tags=tag['id'], page=page)
```

Comment:
We can skip this lookup if the tag doesn't exist and just return posts as an empty array.